lib/tasks/add_ost_apps.rake

```diff
@@ -17,23 +27,30 @@ namespace :accounts do
           password = args[:admin_password]
           # Create application owner if needed
           app_owner_group = Group.find_or_create_by_name('ost_app_admin_group')
-          user = User.find_or_create_by_username('ost_app_admin') do |new_user|
-            new_user.identity = Identity.create do |identity|
-              identity.password = password
-              identity.password_confirmation = password
-              identity.user_id = user.id
-            end
-            new_user.identity.save!
-            app_owner_group.add_owner(user)
+          user = User.find_or_create_by_username('ost_app_admin')
+          identity = Identity.find_or_create_by_user_id(user.id) do |new_identity|
+            new_identity.password = password
+            new_identity.password_confirmation = password
+            new_identity.save!
           end
+          app_owner_group.add_owner(user)
           auth = Authentication.find_or_create_by_uid(user.identity.id.to_s) do |new_auth|
             new_auth.provider = 'identity'
             new_auth.user_id = user.id
             new_auth.save!
           end
           # For each app above, create and assign defaults if it doesn't exist
+          cb_path = "accounts/auth/openstax/callback"
           app_data.each do |app|
-            url = "https://#{app[:prefix]}#{suffix}.openstax.org/accounts/auth/openstax/callback"
+            # If the suffix parameter starts with `http` assume that
```

Comment:
On vagrant installations, we are serving from a non-standard port (port forwarding on host).  In those cases, we need to provide the full-url with the non-standard port.